### PR TITLE
CRM-21246 - Double chaining problem

### PR DESCRIPTION
**Pull request by johanv**
Overview
----------------------------------------
Fix bug when chaining using the $value syntax

Before
----------------------------------------
```
 $result = $this->callAPISuccess('contact', 'get',  array(
       'id' => $this->_cId_a,
       'api.Relationship.get' => array(
         'relationship_type_id' => 1,
         'contact_id_a' => '$value.id',
         'api.Contact.get' => array(
           'id' => '$value.contact_id_b',
         ),
       ),
     );
```
fails to retrieve contact B

After
----------------------------------------
Contact B retrieved, as demonstrated by unit test

Technical Details
---------------------------------------
Adds an exception for the api chaining specific string when doing api chaining replacements to stop the chain ending too soon. Unit test

 * [CRM-21246: Problem chaining Contact.get - Relationship.get - Contact.get](https://issues.civicrm.org/jira/browse/CRM-21246)

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
nice!

File: tests/phpunit/api/v3/RelationshipTest.php [hunk 1]
```diff
@@ -779,6 +779,61 @@ public function testRelationshipsGet() {
   }
 
   /**
+   * Chain Relationship.get and to Contact.get.
```
Comment (johanv) [inline tests/phpunit/api/v3/RelationshipTest.php]: This test passed before applying the fix. I put it inhere while trying to insulate the problem.

File: tests/phpunit/api/v3/RelationshipTest.php [hunk 2]
```diff
@@ -779,6 +779,61 @@ public function testRelationshipsGet() {
...
+
+    $this->assertEquals(1, $result['count']);
+    $relationship = CRM_Utils_Array::first($result['values']);
+    $this->assertEquals(1, $relationship['api.Contact.get']['count']);
+    $contact = CRM_Utils_Array::first($relationship['api.Contact.get']['values']);
+    $this->assertEquals($this->_cId_b, $contact['id']);
+  }
+
+  /**
+   * Chain Contact.get to Relationship.get and again to Contact.get.
```
Comment (johanv) [inline tests/phpunit/api/v3/RelationshipTest.php]: This test failed before. Now it passes.

**Comment by eileenmcnaughton** [comment]
This feels OK to me. I think the api. is a very specific syntax & won't have side effects. Am giving 'ready to merge' label which allows a few days cool down in case @colemanw @totten have any thoughts.

**Comment by seamuslee001** [comment]
I agree with the change here and i agree with Eileen's analysis. Also having a unit test is a big +1 for me.

**Pull request merged**